### PR TITLE
Commit changes from https://github.com/tnierman/old-backplane-tools

### PR DESCRIPTION
**Pull request by tnierman**
Adds changes from the [prototyping repo](https://github.com/tnierman/old-backplane-tools) to the official backplane-tools repo

### CODE REVIEW

File: pkg/tool/ocm/ocm.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+	release, err := t.source.FetchLatestRelease()
+	if err != nil {
+		return err
+	}
+
+	// Determine which assets to download
+	var checksumAsset gogithub.ReleaseAsset
+	var ocmBinaryAsset gogithub.ReleaseAsset
+
+assetLoop:
```
Comment (NautiluX) [inline pkg/tool/ocm/ocm.go]: do we need this label if there is no nested loop?

File: pkg/tool/ocm/ocm.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+	for _, asset := range release.Assets {
+		// Exclude assets that do not match system OS
+		if !strings.Contains(asset.GetName(), runtime.GOOS) {
+			continue assetLoop
```
Comment (NautiluX) [inline pkg/tool/ocm/ocm.go]:
```suggestion
			continue
```

File: pkg/tool/ocm/ocm.go [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+		}
+		// Exclude assets that do not match system architecture
+		if !strings.Contains(asset.GetName(), runtime.GOARCH) {
+			continue assetLoop
```
Comment (NautiluX) [inline pkg/tool/ocm/ocm.go]:
```suggestion
			continue
```

File: pkg/tool/ocm/ocm.go [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+			if checksumAsset.GetName() != "" {
+				return fmt.Errorf("detected duplicate ocm-cli checksum assets")
+			}
+			checksumAsset = asset
+		} else {
+			if ocmBinaryAsset.GetName() != "" {
+				return fmt.Errorf("detected duplicate ocm-cli binary assets")
+			}
+			ocmBinaryAsset = asset
+		}
```
Comment (NautiluX) [inline pkg/tool/ocm/ocm.go]:
```suggestion
			}
			checksumAsset = asset
			continue
		}
		if ocmBinaryAsset.GetName() != "" {
			return fmt.Errorf("detected duplicate ocm-cli binary assets")
		}
		ocmBinaryAsset = asset
		}
```

**Review by NautiluX — CHANGES_REQUESTED** [review]

File: pkg/tool/ocm/ocm.go (hunk 1 above)
Comment (tnierman) [reply pkg/tool/ocm/ocm.go]: I prefer to be extremely specific when using `continue`, but admittedly there's probably no need here.

**Review by mjlshen — COMMENTED** [review]
/lgtm

**Pull request merged**